scripts/nasa/cmip6_sea_level/process.py

```diff
@@ -0,0 +1,312 @@
+            'measuredProperty: dcs:surfaceLevel',
+            'measurementQualifier: dcs:DifferenceRelativeToBaseDate',
+            'baseDate: dcs:CMIP6ReferencePeriod'
+        ],
+        _SVID: ('DifferenceRelativeToCMIP6ReferenceDate_SeaLevel', 1),
+        _NCVAR: 'sea_level_change',
+        _UNIT: 'Millimeter',
+    },
+    'rates': {
+        _PV: [
+            'populationType: dcs:SeaBodyOfWater',
+            'measuredProperty: dcs:surfaceLevel',
+            'measurementQualifier: dcs:RateOfChange'
+        ],
+        _SVID: ('RateOfChange_SeaLevel', 1),
+        _NCVAR: 'sea_level_change_rate',
+        _UNIT: 'MillimeterPerYear',
+    },
+    'ssp119': {
+        _PV: [
+            'socioeconomicScenario: dcs:SSP1', 'emissionsScenario: dcs:RCP1.9'
+        ],
+        _SVID: ('SSP119', 2),
+    },
+    'ssp126': {
+        _PV: [
+            'socioeconomicScenario: dcs:SSP1', 'emissionsScenario: dcs:RCP2.6'
+        ],
+        _SVID: ('SSP126', 2),
+    },
+    'ssp245': {
+        _PV: [
+            'socioeconomicScenario: dcs:SSP2', 'emissionsScenario: dcs:RCP4.5'
+        ],
+        _SVID: ('SSP245', 2),
+    },
+    'ssp370': {
+        _PV: [
+            'socioeconomicScenario: dcs:SSP3', 'emissionsScenario: dcs:RCP7.0'
+        ],
+        _SVID: ('SSP370', 2),
+    },
+    'ssp585': {
+        _PV: [
+            'socioeconomicScenario: dcs:SSP5', 'emissionsScenario: dcs:RCP8.5'
+        ],
+        _SVID: ('SSP585', 2),
+    },
+    'low': {
+        _MMETHOD: 'IPCC_LowConfidence',
+    },
+    'medium': {
+        _MMETHOD: 'IPCC_MediumConfidence',
+    },
+}
+
+
+def _fname(file_path):
+    return os.path.basename(file_path)[:-3]  # strip '.nc'
+
+
+def parse_sv_info(file_path):
+    fname = _fname(file_path)
+    sv_info = {
+        _PV: [],
+    }
+    sv_id_parts = {}
+    for part in fname.split('_'):
+        if part not in _SV_CONFIG:
+            continue
+        part_info = _SV_CONFIG[part]
+
+        if _PV in part_info:
+            sv_info[_PV].extend(part_info[_PV])
+
+        if _SVID in part_info:
+            id_pair = part_info[_SVID]
+            assert id_pair[1] not in sv_id_parts, sv_id_parts
+            sv_id_parts[id_pair[1]] = id_pair[0]
+
+        if _NCVAR in part_info:
+            assert _NCVAR not in sv_info
+            sv_info[_NCVAR] = part_info[_NCVAR]
+
+        if _UNIT in part_info:
+            assert _UNIT not in sv_info
+            sv_info[_UNIT] = part_info[_UNIT]
+
+        if _MMETHOD in part_info:
+            assert _MMETHOD not in sv_info
+            sv_info[_MMETHOD] = part_info[_MMETHOD]
+
+    assert _NCVAR in sv_info, 'Could not map to an ncvar for: ' + fname
+    sv_info[_SVID] = '_'.join([sv_id_parts[i] for i in sorted(sv_id_parts)])
+    return sv_info
+
+
+###
+### Code specific to SeaLevel NC files
+###
+
+
+def to_place_dcid(location, prefix=''):
+    if location == -1:
+        return prefix + 'Earth'
+    elif location < 1000000000:
```

Comment:
consider using a constant for this so it's self documenting

Reply:
Good point.